{
 "cells": [
  {
   "cell_type": "code",
   "execution_count": 48,
   "metadata": {},
   "outputs": [
    {
     "ename": "ImportError",
     "evalue": "Traceback (most recent call last):\n  File \"/usr/local/lib/python3.5/dist-packages/tensorflow/python/__init__.py\", line 61, in <module>\n    from tensorflow.python import pywrap_tensorflow\n  File \"/usr/local/lib/python3.5/dist-packages/tensorflow/python/pywrap_tensorflow.py\", line 28, in <module>\n    _pywrap_tensorflow = swig_import_helper()\n  File \"/usr/local/lib/python3.5/dist-packages/tensorflow/python/pywrap_tensorflow.py\", line 24, in swig_import_helper\n    _mod = imp.load_module('_pywrap_tensorflow', fp, pathname, description)\n  File \"/usr/lib/python3.5/imp.py\", line 242, in load_module\n    return load_dynamic(name, filename, file)\n  File \"/usr/lib/python3.5/imp.py\", line 342, in load_dynamic\n    return _load(spec)\nImportError: libcudart.so.8.0: cannot open shared object file: No such file or directory\n\n\nFailed to load the native TensorFlow runtime.\n\nSee https://github.com/tensorflow/tensorflow/blob/master/tensorflow/g3doc/get_started/os_setup.md#import_error\n\nfor some common reasons and solutions.  Include the entire stack trace\nabove this error message when asking for help.",
     "output_type": "error",
     "traceback": [
      "\u001b[0;31m---------------------------------------------------------------------------\u001b[0m",
      "\u001b[0;31mImportError\u001b[0m                               Traceback (most recent call last)",
      "\u001b[0;32m/usr/local/lib/python3.5/dist-packages/tensorflow/python/__init__.py\u001b[0m in \u001b[0;36m<module>\u001b[0;34m()\u001b[0m\n\u001b[1;32m     60\u001b[0m     \u001b[0msys\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0msetdlopenflags\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0m_default_dlopen_flags\u001b[0m \u001b[0;34m|\u001b[0m \u001b[0mctypes\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mRTLD_GLOBAL\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0;32m---> 61\u001b[0;31m     \u001b[0;32mfrom\u001b[0m \u001b[0mtensorflow\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mpython\u001b[0m \u001b[0;32mimport\u001b[0m \u001b[0mpywrap_tensorflow\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0m\u001b[1;32m     62\u001b[0m     \u001b[0msys\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0msetdlopenflags\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0m_default_dlopen_flags\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n",
      "\u001b[0;32m/usr/local/lib/python3.5/dist-packages/tensorflow/python/pywrap_tensorflow.py\u001b[0m in \u001b[0;36m<module>\u001b[0;34m()\u001b[0m\n\u001b[1;32m     27\u001b[0m             \u001b[0;32mreturn\u001b[0m \u001b[0m_mod\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0;32m---> 28\u001b[0;31m     \u001b[0m_pywrap_tensorflow\u001b[0m \u001b[0;34m=\u001b[0m \u001b[0mswig_import_helper\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0m\u001b[1;32m     29\u001b[0m     \u001b[0;32mdel\u001b[0m \u001b[0mswig_import_helper\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n",
      "\u001b[0;32m/usr/local/lib/python3.5/dist-packages/tensorflow/python/pywrap_tensorflow.py\u001b[0m in \u001b[0;36mswig_import_helper\u001b[0;34m()\u001b[0m\n\u001b[1;32m     23\u001b[0m             \u001b[0;32mtry\u001b[0m\u001b[0;34m:\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0;32m---> 24\u001b[0;31m                 \u001b[0m_mod\u001b[0m \u001b[0;34m=\u001b[0m \u001b[0mimp\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mload_module\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0;34m'_pywrap_tensorflow'\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0mfp\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0mpathname\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0mdescription\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0m\u001b[1;32m     25\u001b[0m             \u001b[0;32mfinally\u001b[0m\u001b[0;34m:\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n",
      "\u001b[0;32m/usr/lib/python3.5/imp.py\u001b[0m in \u001b[0;36mload_module\u001b[0;34m(name, file, filename, details)\u001b[0m\n\u001b[1;32m    241\u001b[0m         \u001b[0;32melse\u001b[0m\u001b[0;34m:\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0;32m--> 242\u001b[0;31m             \u001b[0;32mreturn\u001b[0m \u001b[0mload_dynamic\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mname\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0mfilename\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0mfile\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0m\u001b[1;32m    243\u001b[0m     \u001b[0;32melif\u001b[0m \u001b[0mtype_\u001b[0m \u001b[0;34m==\u001b[0m \u001b[0mPKG_DIRECTORY\u001b[0m\u001b[0;34m:\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n",
      "\u001b[0;32m/usr/lib/python3.5/imp.py\u001b[0m in \u001b[0;36mload_dynamic\u001b[0;34m(name, path, file)\u001b[0m\n\u001b[1;32m    341\u001b[0m             name=name, loader=loader, origin=path)\n\u001b[0;32m--> 342\u001b[0;31m         \u001b[0;32mreturn\u001b[0m \u001b[0m_load\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mspec\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0m\u001b[1;32m    343\u001b[0m \u001b[0;34m\u001b[0m\u001b[0m\n",
      "\u001b[0;31mImportError\u001b[0m: libcudart.so.8.0: cannot open shared object file: No such file or directory",
      "\nDuring handling of the above exception, another exception occurred:\n",
      "\u001b[0;31mImportError\u001b[0m                               Traceback (most recent call last)",
      "\u001b[0;32m<ipython-input-48-a6ad2d37b87a>\u001b[0m in \u001b[0;36m<module>\u001b[0;34m()\u001b[0m\n\u001b[1;32m      8\u001b[0m \u001b[0;32mimport\u001b[0m \u001b[0mrandom\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m      9\u001b[0m \u001b[0;32mimport\u001b[0m \u001b[0msys\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0;32m---> 10\u001b[0;31m \u001b[0;32mimport\u001b[0m \u001b[0mtensorflow\u001b[0m \u001b[0;32mas\u001b[0m \u001b[0mtf\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0m",
      "\u001b[0;32m/usr/local/lib/python3.5/dist-packages/tensorflow/__init__.py\u001b[0m in \u001b[0;36m<module>\u001b[0;34m()\u001b[0m\n\u001b[1;32m     22\u001b[0m \u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m     23\u001b[0m \u001b[0;31m# pylint: disable=wildcard-import\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0;32m---> 24\u001b[0;31m \u001b[0;32mfrom\u001b[0m \u001b[0mtensorflow\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mpython\u001b[0m \u001b[0;32mimport\u001b[0m \u001b[0;34m*\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0m\u001b[1;32m     25\u001b[0m \u001b[0;31m# pylint: enable=wildcard-import\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m     26\u001b[0m \u001b[0;34m\u001b[0m\u001b[0m\n",
      "\u001b[0;32m/usr/local/lib/python3.5/dist-packages/tensorflow/python/__init__.py\u001b[0m in \u001b[0;36m<module>\u001b[0;34m()\u001b[0m\n\u001b[1;32m     70\u001b[0m \u001b[0;32mfor\u001b[0m \u001b[0msome\u001b[0m \u001b[0mcommon\u001b[0m \u001b[0mreasons\u001b[0m \u001b[0;32mand\u001b[0m \u001b[0msolutions\u001b[0m\u001b[0;34m.\u001b[0m  \u001b[0mInclude\u001b[0m \u001b[0mthe\u001b[0m \u001b[0mentire\u001b[0m \u001b[0mstack\u001b[0m \u001b[0mtrace\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m     71\u001b[0m above this error message when asking for help.\"\"\" % traceback.format_exc()\n\u001b[0;32m---> 72\u001b[0;31m   \u001b[0;32mraise\u001b[0m \u001b[0mImportError\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mmsg\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0m\u001b[1;32m     73\u001b[0m \u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m     74\u001b[0m \u001b[0;31m# Protocol buffers\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n",
      "\u001b[0;31mImportError\u001b[0m: Traceback (most recent call last):\n  File \"/usr/local/lib/python3.5/dist-packages/tensorflow/python/__init__.py\", line 61, in <module>\n    from tensorflow.python import pywrap_tensorflow\n  File \"/usr/local/lib/python3.5/dist-packages/tensorflow/python/pywrap_tensorflow.py\", line 28, in <module>\n    _pywrap_tensorflow = swig_import_helper()\n  File \"/usr/local/lib/python3.5/dist-packages/tensorflow/python/pywrap_tensorflow.py\", line 24, in swig_import_helper\n    _mod = imp.load_module('_pywrap_tensorflow', fp, pathname, description)\n  File \"/usr/lib/python3.5/imp.py\", line 242, in load_module\n    return load_dynamic(name, filename, file)\n  File \"/usr/lib/python3.5/imp.py\", line 342, in load_dynamic\n    return _load(spec)\nImportError: libcudart.so.8.0: cannot open shared object file: No such file or directory\n\n\nFailed to load the native TensorFlow runtime.\n\nSee https://github.com/tensorflow/tensorflow/blob/master/tensorflow/g3doc/get_started/os_setup.md#import_error\n\nfor some common reasons and solutions.  Include the entire stack trace\nabove this error message when asking for help."
     ]
    }
   ],
   "source": [
    "%matplotlib inline\n",
    "\n",
    "import gym\n",
    "from gym.wrappers import Monitor\n",
    "import itertools\n",
    "import numpy as np\n",
    "import os\n",
    "import random\n",
    "import sys\n",
    "import tensorflow as tf\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 43,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "image/png": "iVBORw0KGgoAAAANSUhEUgAAANEAAAD8CAYAAADpCEEHAAAABHNCSVQICAgIfAhkiAAAAAlwSFlz\nAAALEgAACxIB0t1+/AAADnxJREFUeJzt3X/sVfV9x/Hna1j9g3YRqyMGcKKjbXDZqCWOZGq6ulok\nTdH9YTFLxc0MTTRpo8uCNdnMkiZbVzFpttFgJMXF+qOzVrNYJmNNzbJhBUtRVBQpRr5BmLiow6YW\neO+P8/mux6/fy/d+7/tc7rmX1yO5ued+zq/Pge/r+zn3fM99X0UEZta7Xxt0B8yGnUNkluQQmSU5\nRGZJDpFZkkNkltS3EElaKmmXpN2SVvdrP2aDpn78nUjSDOAl4LPAPuBp4JqIeL7xnZkNWL9GoouA\n3RGxJyLeAx4AlvdpX2YDdUqftjsHeK32eh/we50WluTbJqyN3oiIs6ZaqF8hmpKkVcCqQe3frAuv\ndrNQv0I0BsyrvZ5b2v5fRKwD1oFHIhtu/XpP9DSwQNJ8SacCK4DH+rQvs4Hqy0gUEUck3Qz8KzAD\nWB8RO/uxL7NB68sl7ml3ooWnc2vWrJn2OrfccktqGxPXn8zEbXazTlY/jiPrBP07bIuIxVMt5DsW\nzJIGdnVu2Ez2my77G7qX0a4NTsRIM0w8EpkleSSyaZtqBD3ZRiqPRGZJHolsSlONLMP63q4pHonM\nkjwSdamJ37bD+ht7WPt9ongkMktyiMySfNuPWWe+7cfsRGjFhYW5c+eedH+gs/br9mfSI5FZkkNk\nluQQmSU5RGZJPYdI0jxJP5T0vKSdkr5c2u+QNCZpe3ksa667Zu2TuTp3BLg1Ip6R9BFgm6RNZd5d\nEfGNfPfM2q/nEEXEfmB/mX5H0gtURRvNTiqNvCeSdC7wSeCp0nSzpB2S1kua1cQ+zNoqHSJJHwYe\nBr4SEW8Da4HzgUVUI9WdHdZbJWmrpK2HDx/OdsNsYFIhkvQhqgDdFxHfA4iIAxFxNCKOAXdTFbf/\ngIhYFxGLI2LxzJkzM90wG6jM1TkB9wAvRMSaWvvZtcWuAp7rvXtm7Ze5Ovf7wJeAZyVtL21fBa6R\ntAgIYC9wQ6qHZi2XuTr3H4AmmfV4790xGz6+Y8EsqRUfhZiKPyZh/dBU7QiPRGZJDpFZkkNkluQQ\nmSU5RGZJDpFZkkNkluQQmSU5RGZJDpFZkkNkluQQmSU5RGZJDpFZkkNklpT+PJGkvcA7wFHgSEQs\nlnQG8CBwLtVHxK+OiP/J7susjZoaif4gIhbVvlVsNbA5IhYAm8trs5HUr9O55cCGMr0BuLJP+zEb\nuCZCFMATkrZJWlXaZpcywwCvA7Mb2I9ZKzVRY+HiiBiT9BvAJkkv1mdGREz2xcYlcKsAZs1ypWEb\nXumRKCLGyvNB4BGqiqcHxos4lueDk6znCqg2ErJlhGeWr1VB0kzgcqqKp48BK8tiK4FHM/sxa7Ps\n6dxs4JGqojCnAN+JiI2SngYeknQ98CpwdXI/Zq2VClFE7AF+d5L2Q8BlmW2bDQvfsWCWNBQVULcs\nXTroLtgI+s+GtuORyCzJITJLcojMkhwisySHyCxpKK7OHfuttwfdBbOOPBKZJTlEZkkOkVmSQ2SW\n5BCZJTlEZklDcYn7zV9/d9BdMOvII5FZkkNkltTz6Zykj1NVOR13HvCXwOnAnwH/Xdq/GhGP99xD\ns5brOUQRsQtYBCBpBjBGVe3nT4C7IuIbjfTQrOWaOp27DHglIl5taHtmQ6Opq3MrgPtrr2+WdC2w\nFbg1W8z+zU+8l1ndbHJvNLOZ9Egk6VTgC8B3S9Na4HyqU739wJ0d1lslaaukrYcPH852w2xgmjid\nuwJ4JiIOAETEgYg4GhHHgLupKqJ+gCug2qhoIkTXUDuVGy8fXFxFVRHVbGSl3hOV0sGfBW6oNX9d\n0iKqb4vYO2Ge2cjJVkA9DHx0QtuXUj0yGzJDce/cd46dM+gu2Ai6vKHt+LYfsySHyCzJITJLcojM\nkhwis6ShuDq38bqNUy7zmaVbTkBPbKRc3syXq3gkMktyiMySHCKzJIfILMkhMktyiMyShuISdzf+\nfeOSQXfBhsznL1/TyHY8EpklOURmSQ6RWVJXIZK0XtJBSc/V2s6QtEnSy+V5VmmXpG9K2i1ph6QL\n+9V5szbodiT6NrB0QttqYHNELAA2l9dQVf9ZUB6rqEpomY2srkIUEU8Cb05oXg5sKNMbgCtr7fdG\nZQtw+oQKQGYjJfOeaHZE7C/TrwOzy/Qc4LXacvtK2/u4eKONikYuLEREUJXIms46Lt5oIyETogPj\np2nl+WBpHwPm1ZabW9rMRlImRI8BK8v0SuDRWvu15SrdEuCt2mmf2cjp6rYfSfcDnwbOlLQP+Cvg\nb4CHJF0PvApcXRZ/HFgG7Abepfq+IrOR1VWIIuKaDrMum2TZAG7KdMpsmPiOBbMkh8gsySEyS3KI\nzJIcIrMkh8gsySEyS3KIzJIcIrMkh8gsySEyS3KIzJIcIrMkh8gsySEyS3KIzJIcIrOkKUPUofrp\n30l6sVQ4fUTS6aX9XEk/l7S9PL7Vz86btUE3I9G3+WD1003Ab0fE7wAvAbfV5r0SEYvK48ZmumnW\nXlOGaLLqpxHxREQcKS+3UJXFMjspNfGe6E+BH9Rez5f0E0k/knRJp5VcAdVGReqb8iTdDhwB7itN\n+4FzIuKQpE8B35d0QUS8PXHdiFgHrAOYN2/etKqnmrVJzyORpOuAzwN/XMpkERG/iIhDZXob8Arw\nsQb6adZaPYVI0lLgL4AvRMS7tfazJM0o0+dRfb3KniY6atZWU57Odah+ehtwGrBJEsCWciXuUuCv\nJf0SOAbcGBETv5LFbKRMGaIO1U/v6bDsw8DD2U6ZDRPfsWCW5BCZJTlEZkkOkVmSQ2SW5BCZJTlE\nZkkOkVmSQ2SW5BCZJTlEZkkOkVmSQ2SW5BCZJTlEZkkOkVmSQ2SW1GsF1DskjdUqnS6rzbtN0m5J\nuyR9rl8dN2uLXiugAtxVq3T6OICkhcAK4IKyzj+OFy4xG1U9VUA9juXAA6V01s+A3cBFif6ZtV7m\nPdHNpaD9ekmzStsc4LXaMvtK2we4AqqNil5DtBY4H1hEVfX0zuluICLWRcTiiFg8c+bMHrthNng9\nhSgiDkTE0Yg4BtzNr07ZxoB5tUXnljazkdVrBdSzay+vAsav3D0GrJB0mqT5VBVQf5zrolm79VoB\n9dOSFgEB7AVuAIiInZIeAp6nKnR/U0Qc7U/Xzdqh0QqoZfmvAV/LdMpsmPiOBbMkh8gsySEyS3KI\nzJIcIrMkh8gsySEyS3KIzJIcIrMkh8gsySEyS3KIzJIcIrMkh8gsySEyS3KIzJJ6Ld74YK1w415J\n20v7uZJ+Xpv3rX523qwNpvxkK1Xxxr8H7h1viIgvjk9LuhN4q7b8KxGxqKkOmrVdNx8Pf1LSuZPN\nkyTgauAzzXbLbHhk3xNdAhyIiJdrbfMl/UTSjyRdkty+Wet1czp3PNcA99de7wfOiYhDkj4FfF/S\nBRHx9sQVJa0CVgHMmjVr4myzodHzSCTpFOCPgAfH20oN7kNlehvwCvCxydZ3BVQbFZnTuT8EXoyI\nfeMNks4a/xYISedRFW/ck+uiWbt1c4n7fuC/gI9L2ifp+jJrBe8/lQO4FNhRLnn/M3BjRHT7jRJm\nQ6nX4o1ExHWTtD0MPJzvltnw8B0LZkkOkVmSQ2SW5BCZJTlEZkkOkVmSQ2SW5BCZJTlEZknZu7gb\n8daMY/zL6f876G7YiNiydGl3Cz7xRCP780hkluQQmSU5RGZJrXhPZNakJRs3drVc1++dpuCRyCzJ\nI5GdtLodsaaiiGhkQ6lOSIPvhNkHbYuIxVMt1M3Hw+dJ+qGk5yXtlPTl0n6GpE2SXi7Ps0q7JH1T\n0m5JOyRdmD8Ws/bq5j3REeDWiFgILAFukrQQWA1sjogFwObyGuAKqgIlC6hKYq1tvNdmLTJliCJi\nf0Q8U6bfAV4A5gDLgQ1lsQ3AlWV6OXBvVLYAp0s6u/Gem7XEtK7OlXLCnwSeAmZHxP4y63Vgdpme\nA7xWW21faTMbSV1fnZP0YapKPl+JiLerMtyViIjpXhyoV0A1G2ZdjUSSPkQVoPsi4nul+cD4aVp5\nPljax4B5tdXnlrb3qVdA7bXzZm3QzdU5AfcAL0TEmtqsx4CVZXol8Git/dpylW4J8FbttM9s9ETE\ncR/AxUAAO4Dt5bEM+CjVVbmXgX8DzijLC/gHqjrczwKLu9hH+OFHCx9bp/rZjQj/sdXsOJr5Y6uZ\nHZ9DZJbkEJklOURmSQ6RWVJbPk/0BnC4PI+KMxmd4xmlY4Huj+c3u9lYKy5xA0jaOkp3L4zS8YzS\nsUDzx+PTObMkh8gsqU0hWjfoDjRslI5nlI4FGj6e1rwnMhtWbRqJzIbSwEMkaamkXaWwyeqp12gf\nSXslPStpu6StpW3SQi5tJGm9pIOSnqu1DW0hmg7Hc4eksfJ/tF3Sstq828rx7JL0uWnvsJtbvfv1\nAGZQfWTiPOBU4KfAwkH2qcfj2AucOaHt68DqMr0a+NtB9/M4/b8UuBB4bqr+U30M5gdUH3lZAjw1\n6P53eTx3AH8+ybILy8/dacD88vM4Yzr7G/RIdBGwOyL2RMR7wANUhU5GQadCLq0TEU8Cb05oHtpC\nNB2Op5PlwAMR8YuI+Bmwm+rnsmuDDtGoFDUJ4AlJ20rtCOhcyGVYjGIhmpvLKej62ul1+ngGHaJR\ncXFEXEhVc+8mSZfWZ0Z13jC0l0GHvf/FWuB8YBGwH7izqQ0POkRdFTVpu4gYK88HgUeoTgc6FXIZ\nFqlCNG0TEQci4mhEHAPu5lenbOnjGXSIngYWSJov6VRgBVWhk6Ehaaakj4xPA5cDz9G5kMuwGKlC\nNBPet11F9X8E1fGskHSapPlUlXt/PK2Nt+BKyjLgJaqrIrcPuj899P88qqs7PwV2jh8DHQq5tPEB\n3E91ivNLqvcE13fqPz0UomnJ8fxT6e+OEpyza8vfXo5nF3DFdPfnOxbMkgZ9Omc29BwisySHyCzJ\nITJLcojMkhwisySHyCzJITJL+j/J/FexoTLYjwAAAABJRU5ErkJggg==\n",
      "text/plain": [
       "<matplotlib.figure.Figure at 0x7efc1e525198>"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "import gym\n",
    "from IPython import display\n",
    "import matplotlib\n",
    "import matplotlib.pyplot as plt\n",
    "%matplotlib inline\n",
    "\n",
    "env = gym.make('Breakout-v0')\n",
    "env.reset()\n",
    "img = plt.imshow(env.render(mode='rgb_array')) # only call this once\n",
    "for _ in range(100):\n",
    "    img.set_data(env.render(mode='rgb_array')) # just update the data\n",
    "    display.display(plt.gcf())\n",
    "    display.clear_output(wait=True)\n",
    "    action = env.action_space.sample()\n",
    "    env.step(action)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 44,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "(array([[[0, 0, 0],\n",
       "         [0, 0, 0],\n",
       "         [0, 0, 0],\n",
       "         ...,\n",
       "         [0, 0, 0],\n",
       "         [0, 0, 0],\n",
       "         [0, 0, 0]],\n",
       " \n",
       "        [[0, 0, 0],\n",
       "         [0, 0, 0],\n",
       "         [0, 0, 0],\n",
       "         ...,\n",
       "         [0, 0, 0],\n",
       "         [0, 0, 0],\n",
       "         [0, 0, 0]],\n",
       " \n",
       "        [[0, 0, 0],\n",
       "         [0, 0, 0],\n",
       "         [0, 0, 0],\n",
       "         ...,\n",
       "         [0, 0, 0],\n",
       "         [0, 0, 0],\n",
       "         [0, 0, 0]],\n",
       " \n",
       "        ...,\n",
       " \n",
       "        [[0, 0, 0],\n",
       "         [0, 0, 0],\n",
       "         [0, 0, 0],\n",
       "         ...,\n",
       "         [0, 0, 0],\n",
       "         [0, 0, 0],\n",
       "         [0, 0, 0]],\n",
       " \n",
       "        [[0, 0, 0],\n",
       "         [0, 0, 0],\n",
       "         [0, 0, 0],\n",
       "         ...,\n",
       "         [0, 0, 0],\n",
       "         [0, 0, 0],\n",
       "         [0, 0, 0]],\n",
       " \n",
       "        [[0, 0, 0],\n",
       "         [0, 0, 0],\n",
       "         [0, 0, 0],\n",
       "         ...,\n",
       "         [0, 0, 0],\n",
       "         [0, 0, 0],\n",
       "         [0, 0, 0]]], dtype=uint8), 0.0, False, {'ale.lives': 5})"
      ]
     },
     "execution_count": 44,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "env.step(2)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python 3",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.5.2"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 2
}
